### PR TITLE
Add ggupset

### PR DESCRIPTION
**Pull request by hexylena**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

image

not 100% sure what to label the axes since this is in our case specifically for DEGs, but can be equally useful for other cases. We could expose those to configuration I guess but maybe it's not necessary?

### CODE REVIEW

**Comment by bgruening** [comment]
@hexylena that looks super useful. Thanks. I think the label are good for a first version - but offering them does not hurt.

**Comment by hexylena** [comment]
Ok, I will add the parameters then. Thanks for the feedback!

File: tools/ggupset/ggupset.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		<data name="upset_degree" format="png" from_work_dir="upset-degree.png" label="${tool.name} on ${on_string}: Upset by degree" />
+		<data name="commonalities" format="tabular" from_work_dir="upset.tsv" label="${tool.name} on ${on_string}: Identified Overlapping Groups" />
+	</outputs>
+	<tests>
+		<test>
+			<param name="input" value="HEP3B,HUH,JHH6,JHH7,SNU"/>
+			<param name="n_int" value="40"/>
+			<param name="width" value="1000"/>
+			<param name="height" value="800"/>
+			<output name="upset_degree" file="upset-degree.png" ftype="png" compare="sim_size" delta="10000"/>
```
Comment (bernt-matthias) [inline tools/ggupset/ggupset.xml:38]: Use a size assertion. Then we do not need to save the files.

**Comment by hexylena** [comment]
sure can do. This would be something good to lint on.

**Comment by hexylena** [comment]
I will pay someone money (it's worth like 200€ to me, for the lifetime cumulative annoyance), if they can just extract the test error into a github bot comment here.

Just so I don't have to
1. go to failed action
2. go to summary because the action doesn't include it
3. download a zip file
4. which I unpack for a single html
5. which I can then use to view the report.

**Review by bernt-matthias — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks Helena!

**Comment by hexylena** [comment]
Danke Björn, Matthias!

**Pull request merged**